source/resources/DatasetResource.js

```diff
@@ -24,7 +26,37 @@ class DatasetResource {
   }
 
   parseCsv(csv) {
-    return csvParseRows(csv, d => [d[0], +d[1]])
+    const features = mapResource.getUniqueFeatureIds()
+    const badMapIds = []
+    const badValueIds = []
+    const parsed = csvParseRows(csv, d => [d[0], +d[1]]).filter(row => {
+      const hasId = features.indexOf(row[0]) > -1
+      if (!hasId) {
+        badMapIds.push(row[0])
+      }
+      if (row[1] <= 0 || isNaN(+row[1])) {
```

Comment:
Wasn't `row[1]` already parsed as an int in `csvParseRows` above?

Also, using `+` to do type casting still seems sneaky to me but I know how u D3 coders roll… ;)